File: C# for VB/4-csharp-visual-basic-dotnet-developers-m4-exercise-files/demos/cs/CSModule4/BankAccount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSModule4
{
    public class TransactionEventArgs : EventArgs
    {
        public string Message { get; set; }

        public TransactionEventArgs(string message)
        {
            this.Message = message;
        }
    }

    public class BankAccount
    {
        public event EventHandler<TransactionEventArgs> LargeWithdrawl;
        public event EventHandler<TransactionEventArgs> WithdrawlExceedsBalance;

        private decimal _balance;

        public BankAccount(decimal balance)
        {
            this._balance = balance;
        }

        public bool Withdraw(decimal amount)
        {
            if (this._balance < amount)
            {
                this.OnWithdrawlExceedsBalance();
                return false;
            }

            this._balance -= amount;

            Console.WriteLine("{0:c} is withdrawn from the acount.", amount);

            if (amount >= 1000)
            {
                this.OnLargeWithdrawl();
            }

            return true;
        }

        private void OnWithdrawlExceedsBalance()
        {
            if (this.WithdrawlExceedsBalance != null)
            {
                this.WithdrawlExceedsBalance(this, 
                    new TransactionEventArgs("Withdrawl denied. Exceeds balance."));
            }
        }

        private void OnLargeWithdrawl()
        {
            if (this.LargeWithdrawl != null)
            {
                this.LargeWithdrawl(this,
                    new TransactionEventArgs("ATTENTION: Large withdrawl - call your teenager and see what's up!"));
            }
        }
    }
}

File: C# for VB/4-csharp-visual-basic-dotnet-developers-m4-exercise-files/demos/vb/VBModule4/BankAccount.vb
Public Class BankAccount

    Public Event LargeWithdrawl As EventHandler(Of TransactionEventArgs)
    Public Event WithdrawlExceedsBalance As EventHandler(Of TransactionEventArgs)

    Private _balance As Decimal

    Public Sub New(balance As Decimal)
        Me._balance = balance
    End Sub

    Public Function Withdraw(amount As Decimal) As Boolean

        If Me._balance < amount Then
            Me.OnWithdrawlExceedsBalance()
            Return False
        End If

        Me._balance -= amount

        Console.WriteLine("{0:c} is withdrawn from the account", amount)

        If amount >= 1000 Then
            Me.OnLargeWithdrawl()
        End If

        Return True

    End Function

    Private Sub OnWithdrawlExceedsBalance()
        RaiseEvent WithdrawlExceedsBalance(Me,
            New TransactionEventArgs("Withdrawl denied. Exceeds balance."))
    End Sub

    Private Sub OnLargeWithdrawl()
        RaiseEvent LargeWithdrawl(Me,
            New TransactionEventArgs("ATTENTION: Large withdrawl - call your teenager and see what's up!"))
    End Sub

End Class